src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,369 @@
+    AttributeAccessInterface(MakeOptional(aEndpointId), CameraAvStreamManagement::Id), mDelegate(aDelegate),
+    mEndpointId(aEndpointId), mFeature(aFeature)
+{
+    mDelegate.SetPushAvStreamTransportServer(this);
+}
+
+PushAvStreamTransportServer::~PushAvStreamTransportServer()
+{
+    // Explicitly set the PushAvStreamTransportServer pointer in the Delegate to null.
+
+    mDelegate.SetPushAvStreamTransportServer(nullptr);
+
+    // Unregister command handler and attribute access interfaces
+    CommandHandlerInterfaceRegistry::Instance().UnregisterCommandHandler(this);
+    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
+}
+
+CHIP_ERROR PushAvStreamTransportServer::Init()
+{
+    LoadPersistentAttributes();
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+bool PushAvStreamTransportServer::HasFeature(Feature feature) const
+{
+    return mFeature.Has(feature);
+}
+
+CHIP_ERROR PushAvStreamTransportServer::ReadAndEncodeCurrentConnections(const AttributeValueEncoder::ListEncodeHelper & encoder)
+{
+    for (const auto & currentConnections : mCurrentConnections)
+    {
+        ReturnErrorOnFailure(encoder.Encode(currentConnections));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PushAvStreamTransportServer::AddStreamTransportConnection(const uint16_t transportConnectionId)
+{
+    mCurrentConnections.push_back(transportConnectionId);
+    auto path = ConcreteAttributePath(mEndpointId, PushAvStreamTransport::Id, Attributes::CurrentConnections::Id);
+    mDelegate.OnAttributeChanged(Attributes::CurrentConnections::Id);
+    MatterReportingAttributeChangeCallback(path);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PushAvStreamTransportServer::RemoveStreamTransportConnection(const uint16_t transportConnectionId)
+{
+    mCurrentConnections.erase(std::remove_if(mCurrentConnections.begin(), mCurrentConnections.end(),
+                                             [&](const uint16_t connectionID) { return connectionID == transportConnectionId; }),
+                              mCurrentConnections.end());
+    auto path = ConcreteAttributePath(mEndpointId, PushAvStreamTransport::Id, Attributes::CurrentConnections::Id);
+    mDelegate.OnAttributeChanged(Attributes::CurrentConnections::Id);
+    MatterReportingAttributeChangeCallback(path);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PushAvStreamTransportServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == PushAvStreamTransport::Id);
+    ChipLogError(Zcl, "Push AV Stream Transport: Reading");
+
+    switch (aPath.mAttributeId)
+    {
+    case FeatureMap::Id:
+        ReturnErrorOnFailure(aEncoder.Encode(mFeature));
+        break;
+
+    case SupportedContainerFormats::Id:
```

Comment:
Why do we need to intercept and override the default attributes in the attribute store? We only need to modify these default fields if we require the values to come from the app, since the default storage doesn't meet our requirements. However, it appears that these attributes aren't being handled specially.